napalm_ios/ios.py

```diff
@@ -135,12 +135,13 @@ def open(self):
                                      **self.netmiko_optional_args)
         # ensure in enable mode
         self.device.enable()
-        if not self.dest_file_system:
-            try:
-                self.dest_file_system = self.device._autodetect_fs()
-            except AttributeError:
-                raise AttributeError("Netmiko _autodetect_fs not found please upgrade Netmiko or "
-                                     "specify dest_file_system in optional_args.")
+
+    def _discover_file_system(self):
+        try:
+            return self.device._autodetect_fs()
+        except AttributeError:
```

Comment:
I know you are copying the previous pattern here, but we probably should do:

```
except Exception:
```

If it fails at this point catch it and re-raise the exception with our message.

I would put our message as:

"Netmiko _autodetect_fs failed to workaround specify dest_file_system in optional_args"

In other words, what I put above was for the case when _autodetect_fs didn't exist in Netmiko (which won't happen now). I think we should convert this to something more general.